src/test/kotlin/com/zepben/evolve/database/sqlite/DatabaseSqliteTest.kt

```diff
@@ -259,6 +267,56 @@ class DatabaseSqliteTest {
         validateSchema(SchemaNetworks.createNameTestServices())
     }
 
+    @Test
+    internal fun `post process fails with unresolved references`() {
+        val builder = TestNetworkBuilder()
+        val ns = builder.network
+        builder
+            .fromPowerTransformer() // b0
+            .toPowerElectronicsConnection {  // pec1
+                TapChangerControl("tcc").also { tcc ->
+                    tcc.addRegulatingCondEq(this)
+                    tcc.terminal = Terminal("test").also { ns.add(it) }
+                    regulatingControl = tcc
+                    ns.add(tcc)
+                }
+            }
+            .build()
+
+        val expectedMetadata = MetadataCollection()
+        val expectedDiagramService = DiagramService()
+        val expectedCustomerService = CustomerService()
+
+        assertThat(
+            DatabaseWriter(SCHEMA_TEST_FILE).save(
+                expectedMetadata,
+                mutableListOf(
+                    ns,
+                    expectedDiagramService,
+                    expectedCustomerService
+                )
+            ),
+            equalTo(true)
+        )
+        // Delete a link to cause an unresolved reference.
+        getConnection("jdbc:sqlite:$SCHEMA_TEST_FILE").createStatement().use { statement ->
+            statement.execute("DELETE FROM tap_changer_controls WHERE mRID = 'tcc'")
+        }
```

Comment:
**todo**: This is causing an error on Windows unless the connection is `use`d, because the file is hogged by the db driver until the connection is closed.

Reply:
Modified code:
```
getConnection("jdbc:sqlite:$SCHEMA_TEST_FILE").use { connection ->
    connection.createStatement().use { statement ->
        statement.execute("DELETE FROM tap_changer_controls WHERE mRID = 'tcc'")
    }
}
```

Reply:
fixed, thanks.